django/db/models/sql/query.py

```diff
@@ -1226,7 +1227,7 @@ def add_annotation(self, annotation, alias, select=True):
 
     @property
     def _subquery_fields_len(self):
-        if self.has_select_fields:
+        if self.has_select_fields and tuple(self.selected) != ("pk",):
             return len(self.selected)
         return len(self.model._meta.pk_fields)
```

Comment:
I wonder if this can subtly break when doing things like `values("pk", "other")` in this case the `len` should be `len(self.model._meta.pk_fields) + 1` and not `len(self.selected)`.

This is a problem with composite fields in general though which we can tackle at a later time.